### PR TITLE
add : authentication

### PR DESCRIPTION
**Pull request by ApurvaSawant11**
In this PR, I've added authentication pages with functionality and Protected Routes 

### Files to check:
- Login.jsx
- Signup.jsx
- auth-context.jsx
- routes folder

### Please review and suggest

### CODE REVIEW

File: src/components/sidebar/sidebar.css [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+#header .pro-sidebar-inner {
+    background-color: transparent;
+}
+#header .pro-sidebar-inner .pro-sidebar-layout {
+    overflow-y: hidden;
+}
+#header .pro-sidebar-inner .pro-sidebar-layout .logotext p {
+    font-size: 20px;
+    padding: 0 20px;
+    color: #000;
```
Comment (MayankKumar10) [inline src/components/sidebar/sidebar.css]: declare color in root directory use from there

File: src/components/sidebar/sidebar.css [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    font-weight: bold;
+}
+#header .pro-sidebar-inner .pro-sidebar-layout ul {
+    padding: 0 5px;
+}
+#header .pro-sidebar-inner .pro-sidebar-layout ul .pro-inner-item {
+    color: #000;
+    margin: 10px 0px;
```
Comment (MayankKumar10) [inline src/components/sidebar/sidebar.css]: don't need to use px with 0

**Review by MayankKumar10 — APPROVED** [review]
Look into these suggestions everything else looks good to me, UI looks good, you can merge it.

File: src/components/sidebar/sidebar.css [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+#header .pro-sidebar-inner .pro-sidebar-layout .active {
+    background-image: linear-gradient(0deg, #fece00 0%, #ffe172 100%);
+    border-radius: 8px;
+}
+#header .logo {
+    padding: 20px;
+}
+
+.pro-sidebar .pro-menu a, .pro-sidebar .pro-menu a:hover {
+    color: black !important;
```
Comment (aayushjain-code) [inline src/components/sidebar/sidebar.css:45]: try to use root colors

**Review by aayushjain-code — APPROVED** [review]
Very nice UI and everything defined well you can use root values for some color and fontStylings

File: src/components/sidebar/sidebar.css (hunk 1 above)
Comment (ApurvaSawant11) [reply src/components/sidebar/sidebar.css]: Yess.... I have made the changes

File: src/components/sidebar/sidebar.css (hunk 2 above)
Comment (ApurvaSawant11) [reply src/components/sidebar/sidebar.css]: Missed it somehow. I have made the changes. Thanks.

File: src/components/sidebar/sidebar.css (hunk 3 above)
Comment (ApurvaSawant11) [reply src/components/sidebar/sidebar.css:45]: I have made the changes 👍🏽

**Pull request merged**